tests/test_security_path_traversal.py

```
@@ -204,3 +204,40 @@ def test_run_tailer_path_validation(tmp_path):
+        mock_state.repo_root = Path("/tmp/mock_repo")
+
+        # Mock request with traversal payload
+        request = CompilePreviewRequest(
+            flow_id="test-flow",
+            step_id="test-step",
+            station_id="test-station",
+            run_id="../../etc/passwd"
+        )
+
+        # Patch SpecCompiler to ensure we don't actually try to compile if validation fails
+        with patch("swarm.spec.compiler.SpecCompiler") as MockCompiler:
```

Ruff (0.14.14)

[error] Local variable `MockCompiler` is assigned to but never used

Remove assignment to unused variable `MockCompiler`

(F841)